jupyterhub/singleuser.py

```diff
@@ -17,34 +17,22 @@
 from tornado import ioloop
 from tornado.web import HTTPError
 
+try:
+    import notebook
+except ImportError:
+    raise ImportError("JupyterHub single-user server requires notebook >= 4.0")
```

Comment:
Can you `except .. as e` and `raise ImportError(...) from e` in case the import error is due to something else ? Should be safe as jupyterhub is py3 only.